src/content/it/2020/javascript.md

```diff
@@ -0,0 +1,782 @@
+    ) }}
+  </figcaption>
+</figure>
+
+Ancora più importante, tutti questi strumenti in genere significano più codice e più tempo di elaborazione.
+
+Osservando specificamente i framework in uso, vediamo che i byte JavaScript mediani per le pagine che li utilizzano variano notevolmente a seconda di _cosa_ viene utilizzato.
+
+Il grafico seguente mostra i byte mediani per le pagine in cui è stato trovato uno dei primi 35 framework più comunemente rilevati, suddivisi per client.
+
+{{ figure_markup(
+  image="frameworks-bytes.png",
+  caption="Il numero mediano di kilobyte JavaScript per pagina dal framework JavaScript.",
+  description="Grafico a barre che mostra il numero medio di kilobyte JavaScript per pagina suddivisi e ordinati in base alla popolarità del framework JavaScript. Il framework più popolare, React, ha una quantità media di JS a 1.328 sulle pagine mobile. Altri framework come RequireJS e Angular hanno un numero elevato di byte JS mediani per pagina. Le pagine con MooTools, Prototype, AMP, RightJS, Alpine.js e Svelte hanno una media inferiore a 500 KB per pagina mobile. Ember.js ha un valore anomalo di circa 1.800 KB di JS mediano per pagina mobile.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=955720480&format=interactive",
+  sheets_gid="1206934500",
+  width="600",
+  height="835",
+  sql_file="frameworks-bytes-by-framework.sql"
+) }}
+
+Su uno dello spettro ci sono framework come React o Angular o Ember, che tendono a fornire molto codice indipendentemente dal client. Dall'altro lato, vediamo framework minimalisti come Alpine.js e Svelte che mostrano risultati molto promettenti. I default sono molto importanti e sembra che, iniziando con default altamente performanti, Svelte e Alpine riescano entrambi (finora&hellip; la dimensione del campione è piuttosto piccola) nel creare un insieme di pagine più leggero.
+
+Otteniamo un'immagine molto simile quando guardiamo il tempo del thread principale per le pagine in cui sono stati rilevati questi strumenti.
+
+{{ figure_markup(
+  image="frameworks-main-thread.png",
+  caption="Il tempo mediano del thread principale per pagina dal framework JavaScript.",
+  description="Grafico a barre che mostra il tempo mediano del thread principale per framework. È difficile notare qualcosa di diverso da Ember.js, il cui tempo mediano del thread principale mobile è di oltre 20.000 millisecondi (20 secondi). Il resto dei framework sono minuscoli in confronto.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=691531699&format=interactive",
+  sheets_gid="160832134",
+  sql_file="main_thread_time_frameworks.sql"
+) }}
+
+Il tempo del thread principale mobile di Ember salta fuori e distorce il grafico con il tempo necessario. Tirarlo fuori rende l'immagine un po' più facile da capire.
```

Comment:
There was some extra text added in #1930 - can you translate this. this line should be:

> Ember's mobile main thread time jumps out and kind of distorts the graph with how long it takes. (I spent some more time looking into this and it appears to be heavily influenced [by one particular platform using this framework inefficiently](https://timkadlec.com/remembers/2021-01-26-what-about-ember/), rather than an underlying problem with Ember itself.) Pulling it out makes the picture a bit easier to understand.

Using Google Auto-translate I get this but feel free to improve the translation:

```suggestion
Il tempo del thread principale mobile di Ember salta fuori e distorce il grafico con il tempo necessario. 
(Ho passato un po 'più di tempo a esaminare questo aspetto e sembra essere fortemente influenzato <a hreflang="en" href="https://timkadlec.com/remembers/2021-01-26-what-about-ember/">da una particolare piattaforma che utilizza questo framework in modo inefficiente</a> , piuttosto che un problema di fondo con Ember stesso.) Tirarlo fuori rende l'immagine un po' più facile da capire.
```